platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1134,4 +1138,37 @@ String getFontFallbackFilePath(int importance, int weightHint) {
         return fontFileParser.getFontFallback(importance, weightHint);
     }
 
+    // Tmp Asset Path
+    // ==============
+    String copyAssetToTmpPath(String path) {
+
+        String tmpDir = mapView.getContext().getExternalCacheDir().getAbsolutePath();
+
+        try {
+            InputStream in = assetManager.open(path);
+            File outFile = new File(tmpDir, path);
+
+            if (!outFile.exists()) {
```

Comment:
So if a file at this path already exists then we just return the path. But we don't know for sure that this file is the asset we're trying to access (the application using tangram also has access to this folder and can put files there). 

It's an unlikely scenario, but it's unfortunately kind of tricky to solve. We could:
 - Compare the contents of the asset with the contents of the file, and create a new file at a different path if they differ.
 - Ask the client application to provide a directory reserved for tangram's use.